pkg/prometheus/promcfg.go

```diff
@@ -2025,3 +2028,124 @@ func (cg *ConfigGenerator) appendAdditionalScrapeConfigs(scrapeConfigs []yaml.Ma
 
 	return append(scrapeConfigs, addlScrapeConfigs...), nil
 }
+
+func (cg *ConfigGenerator) appendScrapeConfigs(
+	slices []yaml.MapSlice,
+	scrapeConfigs map[string]*v1alpha1.ScrapeConfig,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32) []yaml.MapSlice {
+	scrapeConfigIdentifiers := make([]string, len(scrapeConfigs))
+	i := 0
+	for k := range scrapeConfigs {
+		scrapeConfigIdentifiers[i] = k
+		i++
+	}
+
+	// Sorting ensures, that we always generate the config in the same order.
+	sort.Strings(scrapeConfigIdentifiers)
+
+	for _, identifier := range scrapeConfigIdentifiers {
+		slices = append(slices,
+			cg.WithKeyVals("scrapeconfig", identifier).generateScrapeConfig(
+				scrapeConfigs[identifier],
+				apiserverConfig,
+				store,
+				shards,
+			),
+		)
+	}
+
+	return slices
+}
+
+func (cg *ConfigGenerator) generateScrapeConfig(
+	m *v1alpha1.ScrapeConfig,
+	apiserverConfig *v1.APIServerConfig,
+	store *assets.Store,
+	shards int32,
+) yaml.MapSlice {
+	jobName := fmt.Sprintf("scrapeconfig/%s/%s", m.Namespace, m.Name)
+	cfg := yaml.MapSlice{
+		{
+			Key:   "job_name",
+			Value: jobName,
+		},
+	}
+
+	cpf := cg.prom.GetCommonPrometheusFields()
+	labeler := namespacelabeler.New(cpf.EnforcedNamespaceLabel, cpf.ExcludedFromEnforcement, false)
+
+	hTs := true
+	cfg = cg.AddHonorTimestamps(cfg, &hTs)
```

Comment:
You can use `"k8s.io/utils/pointer"` if this is not used somewhere else
```suggestion
	cfg = cg.AddHonorTimestamps(cfg, pointer.Bool(true))
```

Reply:
nice thanks!